src/components/forms/AttributeForm.jsx

```diff
@@ -0,0 +1,125 @@
+    marginLeft: '2rem',
+  },
+  field: {
+    marginLeft: '5rem',
+    marginRight: '5rem',
+    minWidth: '10rem',
+    maxWidth: '40rem',
+  },
+}));
+
+const field = {
+  String: StringField,
+  Boolean: BoolField,
+  DateTime: DateTimeField,
+  Integer: IntegerField,
+  Float: FloatField,
+};
+
+export default function AttributeForm({ records, setRecords, ...props }) {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const completedAttr = () => {
+    let number = 0;
+    for (let attrName of Object.keys(records)) {
+      if (records[attrName] !== null && records[attrName] !== undefined) {
+        number += 1;
+      }
+    }
+    return number;
+  };
+
+  const handleSetValue = (attrName) => (value) => {
+    setRecords({ ...records, [attrName]: value });
+  };
+  const handleClearButton = (attrName) => {
+    if (!props.readOnly[attrName]) {
+      setRecords({ ...records, [attrName]: null });
+    }
+  };
+
+  const clearButton = (attrName) => {
+    return (
+      <InputAdornment position="end">
+        <Tooltip title="Reset">
+          <IconButton
+            onClick={() => {
+              handleClearButton(attrName);
+            }}
+            onMouseDown={(event) => event.preventDefault()}
+          >
+            <ClearIcon />
+          </IconButton>
+        </Tooltip>
+      </InputAdornment>
+    );
+  };
+
+  const keyIcon = () => {
+    return (
+      <Tooltip title={t('modelPanels.internalId', 'Unique Identifier')}>
+        <Key
+          fontSize="small"
+          color="disabled"
+          style={{
+            marginRight: '5rem',
+            marginLeft: '0rem',
+          }}
+        />
+      </Tooltip>
+    );
+  };
+
+  return (
+    <form id="AttributesForm-div-root">
+      <legend className={classes.header}>
+        <Typography variant="h6" component="h1">
+          {' '}
+          {t('modelPanels.model') + ': ' + props.modelName}
+        </Typography>
+        <Typography variant="subtitle1">
+          {completedAttr() +
+            ' / ' +
+            Object.keys(records).length +
+            ' ' +
+            t('modelPanels.completed')}
+        </Typography>
+      </legend>
+      ,
+      {Object.keys(props.dataModel).map((attrName) => {
+        const attrType = props.dataModel[attrName];
+        const InputField = field[attrType];
+        return (
+          <InputField
+            className={classes.field}
+            key={attrName + '-' + attrType + 'Field'}
+            label={attrName}
+            value={records[attrName]}
+            onChange={handleSetValue(attrName)}
+            error={props.errMsg[attrName] ? true : false}
+            helperText={props.errMsg[attrName]}
+            InputProps={{
+              readOnly: props.readOnly[attrName] ? true : false,
+              endAdornment: clearButton(attrName),
+            }}
+            rightIcon={props.readOnly[attrName] ? keyIcon : undefined}
+          />
+        );
+      })}
```

Comment:
The use of `props` here suggests that three different props are being passed to handle attributes: `dataModel`, `records`, `errMsg`, and `readOnly`. Am I correct in that assumption?

I would like to suggest that we discuss this at a meeting. IMHO, the current props API is assuming too much knowledge from the parent (e.g. `dataModel`, `records` ?), when in reality it is just a form that receives `attributes`.

I'd prefer if we could simplify the form properties, so the component API is more ergonomic and a bit more intuitive to use in the parent. Perhaps something like the following:

```tsx
interface Attribute {
  value: AttributeValue;
  type: AttributeType;
  error: string;
}

type AttributeValue = boolean | Date | number | string | AttributeValue[];
type AttributeType = 'Boolean' | 'DateTime' | 'Int' | 'Float' | 'String'  // and so on...

// Could also be an Attribute array, and Attribute have a `name: string` prop
interface Attributes {
  [key: string]: Attribute
}

// In the Record page
const [attributes, setAttributes] = useState<Attributes>({ /* Attributes object */  });

<AttributesForm attributes={attributes} onChange={handleAttributeValueChange} />
```